src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshallerHelpers.cs

```diff
@@ -108,7 +108,7 @@ public static TypeSyntax GetCompatibleGenericTypeParameterSyntax(this TypeSyntax
                 InvocationExpression(
                     MemberAccessExpression(
                         SyntaxKind.SimpleMemberAccessExpression,
-                        ParseName(TypeNames.System_Runtime_InteropServices_Marshal),
+                        TypeSyntaxes.System_Runtime_InteropServices_Marshal,
```

Comment:
Everywhere else, the pattern is `ParseName` morphs to `NameSyntaxes` (and `ParseTypeName` -> `TypeSyntaxes`). Given that we've got ** `ParseName` here**, shouldn't we be using `NameSyntaxes.` here (not `TypeSyntaxes`)? Also at [line 125](https://github.com/dotnet/runtime/pull/90339/files#diff-50daf74f0b239edce54693060759980e3d466d5f11185436a7760ffd3bd0598fR125) and [line 134](https://github.com/dotnet/runtime/pull/90339/files#diff-50daf74f0b239edce54693060759980e3d466d5f11185436a7760ffd3bd0598fR134) in this file and [some](https://github.com/dotnet/runtime/pull/90339/files#diff-f86fa100db6f00fde1b59bfa58c8ecfcc2933a6289f83e111265426de509df55R496) more [other](https://github.com/dotnet/runtime/pull/90339/files#diff-f86fa100db6f00fde1b59bfa58c8ecfcc2933a6289f83e111265426de509df55R372) similar [places](https://github.com/dotnet/runtime/pull/90339/files#diff-f86fa100db6f00fde1b59bfa58c8ecfcc2933a6289f83e111265426de509df55R178) and [here](https://github.com/dotnet/runtime/pull/90339/files#diff-6b4637c9c23e883b3a61f83b56e2ad0b6350afb8452121d793280f5025de8309R50) ... or am I just making an invalid assumption?

Reply:
Either ParseName or ParseTypeName would work here since the method takes an ExpressionSyntax, but since these are all types, I'd prefer to use the TypeSyntaxes class for them. The places that I used NameSyntaxes.<TypeName> require a NameSyntax parameter, basically all just for generating attribute applications.